rmw_fastrtps_cpp/src/rmw_node.cpp

```diff
@@ -127,8 +127,13 @@ create_node(
   node_impl->secondaryPubListener = tnat_2;
 
   edp_readers = participant->getEDPReaders();
-  if (!(edp_readers.first->setListener(tnat_1) & edp_readers.second->setListener(tnat_2))) {
-    RMW_SET_ERROR_MSG("Failed to attach ROS related logic to the Participant");
+  if (edp_readers.first && edp_readers.second) {
```

Comment:
maybe just my personal taste, but I would proof every potentially failing test individually. This makes the error message less ambiguous whether condition 1 or condition 2 fails.  
Also, I like the style of testing for negative first and return/goto fail early. 
```
if (!edp_readers.first) {
  RMW_SET_ERROR_MSG("edp_readers.first is null");
  got fail;
}
...
```

But then again, only my 2 cents.

Reply:
Thanks for the suggestion, why I put them together is that:
1. their definition is a std::pair<> a. k. a std::pair<StatefulReader *, StatefulReader *> edp_readers;
2. getEDPReaders() returns them both NULL at the same time and no case to return one of them NULL

Reply:
Looking at the current documentation of this function [here](https://github.com/eProsima/Fast-RTPS/blob/master/include/fastrtps/participant/Participant.h#L86-L94), one could argue there is no notion that they are always returned NULL - even though this might be the case in the current verison. However, this implementation may change over time, so that we can't certainly rely on this assumption.